### PR TITLE
Added assignment 2 to final presentation

### PR DESCRIPTION
**Pull request by Ramonpwb**

### CODE REVIEW

File: src/power_system_simulation/Final_presentation.ipynb [hunk 1]
```diff
@@ -266,7 +273,459 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 2.1 Power Grid Model (PGM) Input\n",
+    "\n",
+    "The following sections describe a power grid calculation module with the use of the `power-grid-model` library. Firstly, we must handle the following inputs:\n",
+    "- A power grid in PGM format\n",
```
Comment (gijsvanderburgh) [inline src/power_system_simulation/Final_presentation.ipynb:292]: Here a . is missing

File: src/power_system_simulation/Final_presentation.ipynb [hunk 2]
```diff
@@ -266,7 +273,459 @@
...
+    "grid_data = json_deserialize_from_file(\"input_network_data.json\")\n",
+    "active_power_profile = pd.read_parquet(\"active_power_profile.parquet\")\n",
+    "reactive_power_profile = pd.read_parquet(\"reactive_power_profile.parquet\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `grid_data` consists of several different tables containing information about different elements of the grid, like lines, nodes, sources and load information."
```
Comment (gijsvanderburgh) [inline src/power_system_simulation/Final_presentation.ipynb:316]: Is this really a table format?

File: src/power_system_simulation/Final_presentation.ipynb [hunk 3]
```diff
@@ -266,7 +273,459 @@
...
+    "print(reactive_power_profile)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 2.2 Constructing PGM using input data\n",
+    "\n",
+    "Using the previously imported input data, a power grid model (PGM) can be constructed. Furthermore, validation is important in this step, and as such an error should be raised if the data is invalid. The PGM can be constructed by calling the class `pfp.PowerFlow`. "
```
Comment (gijsvanderburgh) [inline src/power_system_simulation/Final_presentation.ipynb:412]: Such an error like what?

File: src/power_system_simulation/Final_presentation.ipynb [hunk 4]
```diff
@@ -266,7 +273,459 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 2.3 PGM Batch update dataset and power flow calculation\n",
+    "\n",
+    "The batch update dataset includes multiple power profiles (active and reactive power) for various nodes in the grid over a specified period. Instead of updating the grid model for each individual timestamp or node separately, the entire set of updates is applied at once, allowing us to perform the power flow calculation for the entire period.\n",
+    "\n",
+    "Within the `batch_powerflow` function, the power flow calculation is also performed. This is done using the Newton-Raphson method, which returns the `output_data` which contains the solution to the power flow calculation. Furthermore, the batch update dataset is also validated within the function.\n",
```
Comment (gijsvanderburgh) [inline src/power_system_simulation/Final_presentation.ipynb:448]: Can you change the parameters?

**Review by gijsvanderburgh — CHANGES_REQUESTED** [review]

File: src/power_system_simulation/Final_presentation.ipynb (hunk 4 above)
Comment (Ramonpwb) [reply src/power_system_simulation/Final_presentation.ipynb:448]: No

File: src/power_system_simulation/Final_presentation.ipynb [hunk 5]
```diff
@@ -189,16 +194,16 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "# Find downstream vertices\n",
+    "# 1.3 Find downstream vertices\n",
     "\n",
-    " Given an `edge id`, return all the vertices which are in the downstream of the edge, with respect to the source vertex. Including the downstream vertex of the edge itself!\n",
+    " Given an `edge_id`, return all the vertices which are in the downstream of the edge, with respect to the source vertex. Including the downstream vertex of the edge itself!\n",
```
Comment (BrittElbers) [inline src/power_system_simulation/Final_presentation.ipynb:199]: Why do you use a !?

**Review by gijsvanderburgh — APPROVED** [review]
Thank you for improving my suggestions.

File: src/power_system_simulation/Final_presentation.ipynb (hunk 5 above)
Comment (BrittElbers) [reply src/power_system_simulation/Final_presentation.ipynb:199]: Thank you for solving

**Review by BrittElbers — APPROVED** [review]

**Pull request merged**